WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewModel.swift

```diff
@@ -62,9 +82,62 @@ final class InPersonPaymentsMenuViewModel {
             self?.showWebView = nil
         })
     }
+
+    /// This method is just a helper for debugging, we may use it for populating different Banner content based on the fetched objects count
+    ///
+    func displayIPPFeedbackBannerIfEligible() {
+        if isCODEnabled {
+            let resultsCount = resultsController.fetchedObjects.count
+            let numberOfTransactions = 10
+
+            print("IPP transactions within 30 days: \(resultsCount)")
+            print(resultsController.fetchedObjects.map {
+                ("OrderID: \($0.orderID) - PaymentMethodID: \($0.paymentMethodID) - DatePaid: \(String(describing: $0.datePaid))")
+            })
+
+            if resultsCount == 0 {
+                // TODO: This option should use a different results controller or predicate, as we're looking for 0 orders historically, not within 30 days.
+                print("0 transactions. Banner 1 shown")
+            } else if resultsCount < numberOfTransactions {
+                print("< 10 transactions within 30 days. Banner 2 shown")
+            } else if resultsCount >= numberOfTransactions {
+                print(">= 10 transactions within 30 days. Banner 3 shown")
+            }
+        }
+    }
+
+    private func fetchIPPTransactions() {
+        do {
+            try resultsController.performFetch()
+        } catch {
+            DDLogError("Error fetching IPP transactions: \(error)")
+        }
+    }
+}
+
+private extension InPersonPaymentsMenuViewModel {
+    /// Results controller that fetches IPP transactions within the last 30 days
+    ///
+    func createRecentIPPOrdersResultsController() -> ResultsController<StorageOrder> {
+        let today = Date()
+        let paymentGateway = Constants.wcpay
+        let thirtyDaysBeforeToday = Calendar.current.date(
+            byAdding: .day,
+            value: -30,
+            to: today
+        )!
```

Comment:
I needed to force-unwrap this variable as the signature returns a `Date?`, however, I can't think of an alternative that we could use here as a default value if we happen to have a date that could not be calculated with the given input (which I don't see happening neither). One alternative could be to just not pass the `datePaid` and fetch everything in the predicate?

Reply:
In https://github.com/woocommerce/woocommerce-ios/pull/8549/commits/0394db4cb5507639019b3b3b84cb8859cf5020fa I removed the force-unwrap in favour of providing a default Date value, better to get no results than to crash.